docs/cudax/accessor_with_properties.rst

```diff
@@ -0,0 +1,137 @@
+.. _libcudacxx-extended-api-accessor-with-properties:
+
+``mdspan`` Accessor with Properties
+===================================
+
+**Accessor with Properties** is a CUDA-specific ``mdspan`` accessor that allows users to exploit advanced memory access capabilities by replacing the ``mdspan`` default behavior.
+
+The accessor allows to specify the following properties:
+
+- **Pointer aliasing policy**, see `Pointer Aliasing Policy`_ for more details
+- **Memory alignment**, see `Memory Alignment`_ for more details
+- **Cache eviction policy**, see `Cache Eviction Policy`_ for more details
+- **Prefetch size**, see `Prefetch Size`_ for more details
+
+Types and Utilities
+-------------------
+
+The **main class** to describe the memory access properties
+
+.. code:: cpp
+
+  template <typename ElementType,
+            typename AliasingPolicy,
+            typename Alignment,
+            typename Eviction,
+            typename Prefetch>
+  class accessor_with_properties;
+
+A **proxy class** to allow distinguishing between load and store operations
+
+.. code:: cpp
+
+  template <typename ElementType,
+            typename Restrict,
+            typename Alignment,
+            typename Eviction,
+            typename Prefetch>
+  class accessor_reference;
+
+A **factory** to simplify the creation of an accessor with properties. It supports a variable number of properties and without enforcing a specific order
+
+.. code:: cpp
+
+  template <typename ElementType, typename... UserProperties>
+  auto make_accessor_with_properties(UserProperties... properties) noexcept;
+
+  template <typename ElementType, typename... UserProperties>
+  auto add_properties(UserProperties... properties) noexcept;
+
+A set of **predefined accessors**:
+
+.. code:: cpp
+
+  cuda::streaming_accessor
+  cuda::cache_all_accessor
+
+Pointer Aliasing Policy
+-----------------------
+
+The *aliasing policy* specifies how pointers are resolved by the compiler when they are used in an expression
+
+- ``aliasing_policy::restrict``: **No aliasing**. Pointers doesn't overlap (**default**)
+- ``aliasing_policy::may_alias``: **May alias**. Pointers may overlap
+
+Memory Alignment
+----------------
+
+Specifies the alignment of the data, see
+:ref:`cuda::aligned_size_t <libcudacxx-extended-api-memory-access-shapes-aligned-size>`. **Default**: ``alignof(ElementType)``
```

Comment:
not sure why this link is not rendered properly everything seems fine to me.

https://github.com/NVIDIA/cccl/blob/5d09f9792bf240bd441d3eab4bc1026622adb061/docs/libcudacxx/extended_api/shapes/aligned_size_t.rst?plain=1#L1

Reply:
actually we might need to expose libcudacxx rst module to the cudax rst module here
https://github.com/NVIDIA/cccl/blob/5d09f9792bf240bd441d3eab4bc1026622adb061/docs/cudax/index.rst?plain=1#L6-L13

Reply:
seems like this is a pattern in the docs of cudax
https://github.com/NVIDIA/cccl/blob/5d09f9792bf240bd441d3eab4bc1026622adb061/docs/cudax/index.rst?plain=1#L19-L23